src/bit_array.cr

```diff
@@ -140,18 +140,18 @@ struct BitArray
       bits = @bits[0]
 
       bits >>= start
-      bits &= (1 << count) - 1
+      bits &= (1 << count) &- 1
 
       BitArray.new(count).tap { |ba| ba.@bits[0] = bits }
     elsif size <= 64
       # Original fits in int64, we can use bitshifts
       bits = @bits.as(UInt64*)[0]
 
       bits >>= start
-      bits &= (1 << count) - 1
+      bits &= (1 << count) &- 1
 
       if count <= 32
-        BitArray.new(count).tap { |ba| ba.@bits[0] = bits.to_u32 }
+        BitArray.new(count).tap { |ba| ba.@bits[0] = bits.to_u32! }
```

Comment:
I don't see when this can overflow if `bits &= (1 << count) &- 1` is working correctly. However, it is not.

`(1 << count)` is simply zero when `count >= 32`, because `1` is an int32. This doesn't matter though, since we don't care about the MSB bits being zero. So I think the `bits &= (1 << count) &- 1` line can be removed and the `to_u32!` can stay. And I think the masking in the `size <=32` case can be removed too. I'd like a comment left about leaving the trailing upper though.

Usages of `(1 << count)` in the rest of the file must be audited though. I think they're all fine, but i'd like a second pair of eyes.